packages/sitecore-jss-nextjs/src/middleware/redirects-middleware.ts

```diff
@@ -188,20 +213,37 @@ export class RedirectsMiddleware extends MiddlewareBase {
   private async getExistsRedirect(
     req: NextRequest,
     siteName: string
-  ): Promise<(RedirectInfo & { matchedQueryString?: string }) | undefined> {
-    const redirects = await this.redirectsService.fetchRedirects(siteName);
+  ): Promise<RedirectResult | undefined> {
     const { pathname: targetURL, search: targetQS = '', locale } = this.normalizeUrl(
       req.nextUrl.clone()
     );
+    const normalizedPath = targetURL.replace(/\/*$/gi, '');
+    const redirects = await this.redirectsService.fetchRedirects(siteName);
     const language = this.getLanguage(req);
     const modifyRedirects = structuredClone(redirects);
+    let matchedQueryString: string | undefined;
 
     return modifyRedirects.length
-      ? modifyRedirects.find((redirect: RedirectInfo & { matchedQueryString?: string }) => {
+      ? modifyRedirects.find((redirect: RedirectResult) => {
+          if (isRegexOrUrl(redirect.pattern) === 'url') {
+            const parseUrlPattern = redirect.pattern.endsWith('/')
+              ? redirect.pattern.slice(0, -1).split('?')
+              : redirect.pattern.split('?');
+
+            return (
+              (parseUrlPattern[0] === normalizedPath ||
+                parseUrlPattern[0] === `/${locale}${normalizedPath}`) &&
+              areURLSearchParamsEqual(
+                new URLSearchParams(parseUrlPattern[1] ?? ''),
+                new URLSearchParams(targetQS)
+              )
+            );
+          }
+
           // Modify the redirect pattern to ignore the language prefix in the path
           // And escapes non-special "?" characters in a string or regex.
-          redirect.pattern = this.escapeNonSpecialQuestionMarks(
-            redirect.pattern.replace(RegExp(`^[^]?/${language}/`, 'gi'), '')
+          redirect.pattern = escapeNonSpecialQuestionMarks(
```

Comment:
So in this function `escapeNonSpecialQuestionMarks` you identify if the string is `regex` using a simple condition
```js
input.startsWith('/') && input.endsWith('/')
```
Why we can't do the same in `isRegexOrUrl` function and simplify that instead of testing if string is url-like using a complex reg exp?

Reply:
I modified the `escapeNonSpecialQuestionMarks` function by removing the condition that checks whether the input is a regex or a string, as this method is only used for regular expressions.